src/query/functions/temporal/functions.go

```diff
@@ -70,28 +71,28 @@ type functionNode struct {
 	comparisonFunc comparisonFunc
 }
 
-func (f *functionNode) Process(values []float64) float64 {
-	if len(values) == 0 {
+func (f *functionNode) Process(datapoints ts.Datapoints) float64 {
+	if len(datapoints) == 0 {
 		return math.NaN()
 	}
 
 	allNaNs := true
 	result := 0.0
-	prev := values[0]
+	prev := datapoints[0].Value
```

Comment:
Any chance these can be nil?

Reply:
we already have a length check ?

Reply:
I mean datapoints[0] could be nil, so you might hit NPE when you call .Value